pkg/git_provider/gitlab.go

```diff
@@ -0,0 +1,356 @@
+package git_provider
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"strings"
+
+	"github.com/quickube/piper/pkg/conf"
+	"github.com/quickube/piper/pkg/utils"
+
+	"github.com/xanzy/go-gitlab"
+)
+
+type GitlabClientImpl struct {
+	client *gitlab.Client
+	cfg    *conf.GlobalConfig
+}
+
+func NewGitlabClient(cfg *conf.GlobalConfig) (Client, error) {
+	ctx := context.Background()
+
+	client, err := gitlab.NewClient(cfg.GitProviderConfig.Token)
```

Comment:
add URL to GitProviderConfig 
Here is an example:
https://github.com/xanzy/go-gitlab/blob/a62c545478735821bc4b24dcbacfbff69534fa9c/README.md?plain=1#L125